systemd/journal.py

```diff
@@ -553,9 +553,11 @@ class JournalHandler(_logging.Handler):
     the `sender_function` parameter.
     """
 
-    def __init__(self, level=_logging.NOTSET, sender_function=send, **kwargs):
+    def __init__(self, level=_logging.NOTSET, extra=None, sender_function=send, **kwargs):
         super(JournalHandler, self).__init__(level)
 
+        if isinstance(extra, dict):
+            kwargs = extra
```

Comment:
The patch looks good, with the exception of this part. `extra` completely overrides `kwargs`. I think it would be better to simply look at both:
```python
for d in (extra, kwargs):
   for name in d:
       if not _valid_field_name(name):
             ...
       ...
```

Reply:
I see.
I updated to look at both by updating ``kwargs``.
https://github.com/systemd/python-systemd/pull/52/files#diff-fbdb822e1b0f59f5302395f64a7fda54R560

Added test: https://github.com/systemd/python-systemd/pull/52/files#diff-900add079f3a059ccbc90179aeec1511R113